
<html>
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- Copyright 2013 by Arthur A. Gleckler. All rights reserved. -->
    <title>Source for ticket #431</title>
    <link href="/trac.css" rel="stylesheet" type="text/css"></head>
  <body>
    <header>This site is a static rendering of the <a href="https://trac.edgewall.org/">Trac</a> instance that was used by <a href="http://scheme-reports.org/2015/working-group-1.html">R7RS-WG1</a> for its work on R7RS-small (<a href="/attachment/r7rs.pdf">PDF</a>), which was ratified in 2013. For more information, see <a href="/">Home</a>.</header>
    <ul class="navigation">
      <li><a href="/">Home</a></li>
      <li><a href="/ticket/">All tickets</a></li>
      <li><a href="/wiki/">All wiki pages</a></li></ul>
    <h1>Source for ticket #431</h1>
    <h2>cc</h2>
    <pre></pre>
    <h2>changetime</h2>
    <pre>2012-07-02 05:31:22</pre>
    <h2>component</h2>
    <pre>WG1 - Core</pre>
    <h2>description</h2>
    <pre>Submitter's name: Richard Kelsey

Submitter's email: kelsey at s48.org

Relevant draft: r7rs draft 6

Type: clarification

Priority: minor

Relevant section of draft: Derived expressions

Summary: The continuation used when 'guard' re-raises an exception isn't specified.

Page 19, Exception Handling (Handling should be lower case, by the way)

   &quot;If every cond's test evaluates to #f and there is no else clause, then raise-continuable is re-invoked on the raised object within the dynamic environment of the original call to raise except that the current exception handler is that of the guard expression.&quot;

I think that the intent is that the raise-continuable have the same continuation as the handler installed by guard, but it doesn't actually say this.  Also, the original call might have been raise or raise-continuable.  I suggest changing this to:

   &quot;If every cond's test evaluates to #f and there is no else clause, then raise-continuable is invoked on the raised object with the continuation and dynamic environment of the installed exception handler's invocation, except that the current exception handler is that of the guard expression.&quot;
</pre>
    <h2>id</h2>
    <pre>431</pre>
    <h2>keywords</h2>
    <pre></pre>
    <h2>milestone</h2>
    <pre></pre>
    <h2>owner</h2>
    <pre>alexshinn</pre>
    <h2>priority</h2>
    <pre>major</pre>
    <h2>reporter</h2>
    <pre>cowan</pre>
    <h2>resolution</h2>
    <pre>fixed</pre>
    <h2>severity</h2>
    <pre></pre>
    <h2>status</h2>
    <pre>closed</pre>
    <h2>summary</h2>
    <pre>Formal Comment: continuation used when 'guard' re-raises an exception</pre>
    <h2>time</h2>
    <pre>2012-07-02 02:22:13</pre>
    <h2>type</h2>
    <pre>defect</pre>
    <h1>Changes</h1>
    <h2>Change at time 2012-07-02 05:31:22</h2>
    <h2>author</h2>
    <pre>cowan</pre>
    <h2>field</h2>
    <pre>comment</pre>
    <h2>newvalue</h2>
    <pre>The reference to `raise-continuable` was apparently an editorial error, and has been changed to `raise`, which is what R6RS (and SRFI 34) say.</pre>
    <h2>oldvalue</h2>
    <pre>1</pre>
    <h2>raw-time</h2>
    <pre>1341181882465053</pre>
    <h2>ticket</h2>
    <pre>431</pre>
    <h2>time</h2>
    <pre>2012-07-02 05:31:22</pre>
    <h2>Change at time 2012-07-02 05:31:22</h2>
    <h2>author</h2>
    <pre>cowan</pre>
    <h2>field</h2>
    <pre>resolution</pre>
    <h2>newvalue</h2>
    <pre>fixed</pre>
    <h2>oldvalue</h2>
    <pre></pre>
    <h2>raw-time</h2>
    <pre>1341181882465053</pre>
    <h2>ticket</h2>
    <pre>431</pre>
    <h2>time</h2>
    <pre>2012-07-02 05:31:22</pre>
    <h2>Change at time 2012-07-02 05:31:22</h2>
    <h2>author</h2>
    <pre>cowan</pre>
    <h2>field</h2>
    <pre>status</pre>
    <h2>newvalue</h2>
    <pre>closed</pre>
    <h2>oldvalue</h2>
    <pre>new</pre>
    <h2>raw-time</h2>
    <pre>1341181882465053</pre>
    <h2>ticket</h2>
    <pre>431</pre>
    <h2>time</h2>
    <pre>2012-07-02 05:31:22</pre></body></html>